HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,736 @@
+    * auto-association slots behave as though receiver-sig-required is true for
+    * airdrops).
+    */
+    rpc claimAirdrop (Transaction) returns (TransactionResponse);
+
+    /**
+    * Reject tokens in balance
+    */
+    rpc rejectToken (Transaction) returns (TransactionResponse);
+}
+```
+
+### Account Airdrop Defaults
+
+The introduction of airdrop support status requires accounts to specify their level of appetite for airdrops at account creation time. 
+
+To support potential varied DeFi personas the default status will rely on the account creation type
+
+- User-created accounts
+    - If the maximum  `maxAutomaticAssociations` value is present in the transaction there is no need for the ledger to choose the default configuration for these accounts. Wallets and SDK users can control whether they want to create `frictionless airdrop enabled` accounts or airdrop disabled accounts.
+    - If the `maxAutomaticAssociations` value is not present in the transaction the network will choose conservatively to make these accounts `spam-free`.
+- Auto-created accounts
+    - Any auto-created account (such as a hollow account) will default to be `frictionles-airdrop enabled` in which `maxAutomaticAssociations` is set to -1.
+
+Contracts by default will be spam-free in nature. To create a `frictionles-airdrop enabled` contract, the `maxAutomaticAssociations` can be set during a `ContractCreate` HAPI transaction and will require a valid admin key to ensure balance and associations can be managed.
+
+In future the network could adopt a system contract function that supports maxAutoAssociation changes
+
+### System Contract Functions
+
+HTS system contracts will need to support the new airdrop related transactions to ensure the feature is exposed via smart contract functions. Additionally, redirect functions will be provided to ensure optimal UX in which EOAs can initiate transactions through a DApp that affect their account.
+
+| selector | function | IHRC |
+| --- | --- | --- |
+| 0x… | <htsSystemContractAddress>.airdropTokens(TokenTransferList[] memory tokenTransfers) |  |
+| 0x… | <htsTokenAddress>.cancelAirdropFT(address receiverAddress) | Y |
+| 0x… | <htsTokenAddress>.cancelAirdropNFT(address receiverAddress, int64 serialNumber) | Y |
+| 0x… | <htsSystemContractAddress>.cancelAirdrops(TokenPendingAirdrop[] memory pendingAirdrops) |  |
+| 0x… | <htsTokenAddress>.claimAirdropFT(address senderAddress) | Y |
+| 0x… | <htsTokenAddress>.claimAirdropNFT(address senderAddress, int64 serialNumber) | Y |
+| 0x… | <htsSystemContractAddress>.claimAirdrops(TokenPendingAirdrop[] memory pendingAirdrops) |  |
+| 0x… | <htsTokenAddress>.rejectTokenFT() | Y |
+| 0x… | <htsTokenAddress>.rejectTokensNFT(int64[] serialNumber) | Y |
+| 0x… | <htsSystemContractAddress>.rejectTokens(address[] memory ftAddresses, NftId[] memory nftIds) |  |
+| 0x… | <hasSystemContractAddress>.setAutomaticAssociations(boolean enableAutoAssociations) | Y |
```

Comment:
Is the idea here to allow `autoAssociations` on the system contract itself or to a specific account?